docs/howto/openstack/cinder/retype-volumes.md

```diff
@@ -0,0 +1,134 @@
+You must thus detach it first.
+It is safest to do this while the server is shut down:
+
+```console
+$ openstack server stop testsrv
+
+$ openstack server list -c Name -c Status
++---------+---------+
+| Name    | Status  |
++---------+---------+
+| testsrv | SHUTOFF |
++---------+---------+
+```
+
+Once your server is in the `SHUTOFF` state, you can safely proceed to detaching the volume.
+This will change the volume status from `in-use` to `available`.
+
+```console
+$ openstack server remove volume testsrv testvol
+
+$ openstack volume list
++--------------------------------------+---------+-----------+------+-------------+
+| ID                                   | Name    | Status    | Size | Attached to |
++--------------------------------------+---------+-----------+------+-------------+
+| e233e7f3-f33b-4d7a-8f5b-785b34f670bf | testvol | available |   50 |             |
++--------------------------------------+---------+-----------+------+-------------+
+```
+
+## Retyping the volume
+
+With the volume safely detached, you can now change its volume type.
+If you were to change the volume type from `default` to `ceph_hdd`, you would proceed as follows:
+
+```console
+$ openstack volume set --type ceph_hdd --retype-policy on-demand testvol
+
+$ openstack volume list --long
++---------------+---------+----------+------+---------+----------+-------------+------------+
+| ID            | Name    | Status   | Size | Type    | Bootable | Attached to | Properties |
++---------------+---------+----------+------+---------+----------+-------------+------------+
+| e233e7f3-f33b | testvol | retyping |   50 | default | false    |             |            |
+| -4d7a-8f5b-78 |         |          |      |         |          |             |            |
+| 5b34f670bf    |         |          |      |         |          |             |            |
++---------------+---------+----------+------+---------+----------+-------------+------------+
+```
+
+Note that the volume status changes from `available` to `retyping`: this status change kicks off the actual data migration, which might take a significant amount of time.
+
+## Re-attaching the volume
+
+If you attempt to re-attach the volume while it is still retyping, you will receive an error:
+
+```console
+$ openstack server add volume testsrv testvol
+BadRequestException: 400: Client Error for url: https://sto2.{{brand_domain}}:8774/v2.1/servers/b6a26b0e-f911-4ea2-8e45-51f16442da03/os-volume_attachments, Invalid input received: Invalid volume: Volume e233e7f3-f33b-4d7a-8f5b-785b34f670bf status must be available or downloading to reserve, but the current status is retyping. (HTTP 400) (Request-ID: req-36c4f2e5-ef2f-4ff4-b912-0baed2594f4d)
+```
+
+You must now wait until the volume status changes back from `retyping` to `available`.
+One way to do this is with a bash `until` loop:
+
+```bash
+until [ `openstack volume show -f value -c status testvol` = "available" ]; do
```

Comment:
Maybe it would be more convenient for users to have that as oneliner? that's more readable from other side.